airbyte-cdk/python/airbyte_cdk/sources/declarative/decoders/json_decoder.py

```diff
@@ -17,8 +18,43 @@ class JsonDecoder(Decoder):
 
     parameters: InitVar[Mapping[str, Any]]
 
-    def decode(self, response: requests.Response) -> Union[Mapping[str, Any], List[Any], Any]:
+    def decode(self, response: requests.Response) -> Generator[Mapping[str, Any], None, None]:
         try:
-            return response.json()
+            body_json = response.json()
+            if isinstance(body_json, dict):
```

Comment:
If the body does not parse into a dict, what should happen? Is there a situation where the response is a valid json that doesn't parse into a dict, that won't throw an error in response.json() previously? Should we handle this error case explicitly?

Reply:
> If the body does not parse into a dict, what should happen?

In this case I expect body to be an `array`. This block is added to align with protocol and always return iterable object: 
if body is array -> yield from array; if it is not (likely object, but can be any simple type as well) -> put it inside the array and yield from array


> Is there a situation where the response is a valid json that doesn't parse into a dict, that won't throw an error in response.json() previously?

Response can be a simple `string`/`number` (and `.json()` won't throw any exception), but I can't remember the last time I saw this.

> Should we handle this error case explicitly?

Strictly speaking, it is not an error but rather an edge case. It is not handled in any way now, and we have not seen any problems so far; thus, I'd prefer to leave it as is.